src/dodal/cli.py

```diff
@@ -0,0 +1,55 @@
+import click
+
+from dodal.beamlines import all_beamline_names, module_name_for_beamline
+from dodal.utils import make_all_devices
+
+from . import __version__
+
+
+@click.group(invoke_without_command=True)
+@click.version_option(version=__version__, message="%(version)s")
+@click.pass_context
+def main(ctx: click.Context) -> None:
+    if ctx.invoked_subcommand is None:
+        print("Please invoke subcommand!")
+
+
+@main.command(name="connect")
+@click.argument(
+    "beamline",
+    type=click.Choice(list(all_beamline_names())),
+    required=True,
+)
+@click.option(
+    "-a",
+    "--all",
+    is_flag=True,
+    help="Attempt to connect to devices marked as skipped",
+    default=False,
+)
+@click.option(
+    "-s",
+    "--sim-backend",
+    is_flag=True,
+    help="Connect to a sim backend, this initializes all device objects but does not "
+    "attempt any I/O. Useful as a a dry-run.",
+    default=False,
+)
+def connect(beamline: str, all: bool, sim_backend: bool) -> None:
+    """Initialises a beamline module, connects to all devices, reports
+    any connection issues."""
+
+    module_name = module_name_for_beamline(beamline)
+
+    from bluesky import RunEngine
```

Comment:
Nit: Does pylance not give you an error here about `RunEngine` not being exported from `bluesky`? I have to use `from bluesky.run_engine import RunEngine` to fix

Reply:
Interesting, yes it does, even though `RunEngine` is exported from the top level: https://github.com/bluesky/bluesky/blob/27ca36cc60b718ad3f2f2bb343b1d086aec77708/src/bluesky/__init__.py#L4

I suspect pylance is only looking for `__all__`, will change it to reduce the nasty red lines.

Reply:
Also unsure why I'm doing the import inside the function, will change that too